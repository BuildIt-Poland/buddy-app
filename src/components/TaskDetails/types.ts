import { RouteComponentProps } from 'react-router-dom';
import { SnackBarProps } from 'decorators/withSnackBar';

export interface TaskDetailsProps extends RouteComponentProps, SnackBarProps {}
