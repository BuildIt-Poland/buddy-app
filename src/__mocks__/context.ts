import { AuthContextData } from 'context/AuthStore';

export const authContext = (context?: Partial<AuthContextData>) => ({
  data: { role: 'BUDDY', token: 'dummy-token', userId: '1' },
  loading: false,
  error: null,
  isAuthenticated: true,
  login: jest.fn(),
  logout: jest.fn(),
  ...context,
});
