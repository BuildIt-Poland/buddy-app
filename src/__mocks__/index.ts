export * from './general';
export * from './graphql';
export * from './schema';
