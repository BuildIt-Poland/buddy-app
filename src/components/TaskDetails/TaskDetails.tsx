import React from 'react';
import { useParams } from 'react-router-dom';

const TaskDetails: React.FC = () => {
  const { taskId } = useParams();

  return (
    <div data-testid='task-details-page'>
      <h1>Task Details</h1>
      <p>ID: {taskId}</p>
    </div>
  );
};

export default TaskDetails;
