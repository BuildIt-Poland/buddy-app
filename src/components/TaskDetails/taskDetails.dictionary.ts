interface TaskDetails {
  COMPLETED: string;
  UNCOMPLETED: string;
  SUCCESS_MESSAGE: string;
  ERROR_MESSAGE: string;
}

const DICTIONARY: TaskDetails = {
  COMPLETED: 'Completed',
  UNCOMPLETED: 'Uncompleted',
  SUCCESS_MESSAGE: 'Task status updated',
  ERROR_MESSAGE: 'An error ocurred updating task',
};

export default DICTIONARY;
