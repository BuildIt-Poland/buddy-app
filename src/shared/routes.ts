enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  ROUTE_404 = '/404',
  BUDDY_ADD_TASK = '/buddy/newbies/:newbieId/add-task',
  BUDDY_TASK_DETAILS = '/buddy/newbies/:newbieId/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/newbies/:newbieId/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/newbies',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbies/:newbieId/details',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
  NEWBIE_BUDDY_DETAILS = '/newbie/buddy/:buddyId/details',
}

export { ROUTES };
