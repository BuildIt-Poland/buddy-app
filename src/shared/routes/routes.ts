enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  ERROR = '/404',
  BUDDY_ADD_TASK = '/buddy/add-task',
  BUDDY_TASK_DETAILS = '/buddy/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/select-newbie',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbie/:id/details',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
}

export { ROUTES };
