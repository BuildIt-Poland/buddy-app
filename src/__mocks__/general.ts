import { ROUTES } from 'shared/routes';

export const mockLocation = (path: string = ROUTES.BASE) => ({
  key: 'utwyk7',
  pathname: path,
});

export const mockSchemaTypes = (types: any = {}) => ({
  UserRole: {
    Newbie: 'NEWBIE',
    Buddy: 'BUDDY',
  },
  TaskStatus: {
    Completed: 'COMPLETED',
    Uncompleted: 'UNCOMPLETED',
  },
  ...types,
});
