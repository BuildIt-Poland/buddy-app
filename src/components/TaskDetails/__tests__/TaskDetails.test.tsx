import React from 'react';
import { create, act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import { MemoryRouter } from 'react-router';
import { ROUTES } from 'shared/routes';
import { taskDetailsMock } from '__mocks__';
import waitForExpect from 'wait-for-expect';
import TaskDetails from '../TaskDetails';

jest.mock('components/PageContainer', () => 'PageContainer');
jest.mock('components/Header', () => 'Header');
jest.mock('@material-ui/core/CircularProgress', () => 'CircularProgress');
jest.mock('components/TaskCheckbox', () => 'TaskCheckbox');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('decorators/withSnackBar', () => (component: React.FC<any>) => component);

describe('Component - TaskDetails', () => {
  test('renders correctly', async () => {
    const component = create(
      <MockedProvider mocks={taskDetailsMock} addTypename={false}>
        <MemoryRouter initialEntries={[ROUTES.BASE]}>
          <TaskDetails />
        </MemoryRouter>
      </MockedProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();

    await act(async () => {
      await waitForExpect(() => {
        expect(component.toJSON()).toMatchSnapshot();
      });
    });
  });
});
