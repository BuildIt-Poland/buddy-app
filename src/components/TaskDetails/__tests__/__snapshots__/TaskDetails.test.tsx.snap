// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TaskDetails renders correctly 1`] = `
<AppWrapper
  data-testid="task-details-page"
>
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorSecondary makeStyles-progress-4 MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
      style={Object {}}
    />
  </div>
  <NavBar
    onClick={[Function]}
    type="back"
  />
  <svg
    className="makeStyles-background-28"
  >
    <ellipse
      cx="70%"
      cy="300"
      rx="75%"
      ry="227"
    />
  </svg>
</AppWrapper>
`;
